sec2_the-ruby-programming-language/2.9_authenticator.rb

```diff
@@ -0,0 +1,39 @@
+users = [
+        {username: "michelle", password: "password1"},
+        {username: "apple", password: "password2"},
+        {username: "data", password: "password3"},
+        {username: "delta", password: "password4"},
+        {username: "alpha", password: "password5"},
+        ]
+
+def auth_user(username, password, list_of_users)
+  list_of_users.each do |user_record|
+    if user_record[:username] == username && user_record[:password] == password
+      return user_record
+    end
+  end
+  "Credentials were not correct"
+end
```

Comment:
Line 15, no need to put `p`, `print` or `puts` because if the loop does not return anything, it will return the string in Line 15 implicitly

Reply:
in what cases will return implicitly?

Reply:
Return the last statement in the function when 
1. no explicit return is called (that means there is no explicit `return` in the code) or 
2. explicit return did not manage to be called (that means there is explicit `return` in the code but no execution is done due to some conditions)